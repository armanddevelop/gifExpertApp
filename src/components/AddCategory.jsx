import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    inputValue.trim().length > 3 &&
      setCategories((category) => [inputValue, ...category]);
    setInputValue("");
  };
  return (
    <form onSubmit={handleSubmit} noValidate>
      <input
        placeholder="Search for a Gif"
        type="text"
        value={inputValue}
        onChange={(e) => handleChange(e)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
