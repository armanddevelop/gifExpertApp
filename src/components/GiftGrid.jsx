import { useFetchGift } from "../Hooks/useFetchGif";
import { GiftGridItem } from ".";
export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGift(category);
  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Cargando.....</p>
      )}
      <div className="card-grid">
        {images.map((image) => (
          <GiftGridItem key={image.id} img={image} />
        ))}
      </div>
    </>
  );
};
