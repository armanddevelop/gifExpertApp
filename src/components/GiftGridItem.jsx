export const GiftGridItem = ({ img }) => {
  const { title, url, username } = img;
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={username} />
      <p>{title === "" ? "no title" : title}</p>
    </div>
  );
};
