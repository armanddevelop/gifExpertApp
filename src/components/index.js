export { GiftGridItem } from "./GiftGridItem";
export { GiftGrid } from "./GiftGrid";
export { AddCategory } from "./AddCategory";
