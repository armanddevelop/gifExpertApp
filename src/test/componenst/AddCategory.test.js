import { AddCategory } from "../../components";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Test to AddCategory component", () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);
  it("snapshot for the AddCategory component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
