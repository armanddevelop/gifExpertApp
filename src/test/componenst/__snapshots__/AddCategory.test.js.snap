// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to AddCategory component snapshot for the AddCategory component 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search for a Gif"
    type="text"
    value=""
  />
</form>
`;
